src/backend/distributed/transaction/relation_access_tracking.c

```diff
@@ -20,14 +20,27 @@
 #include "access/xact.h"
 #include "distributed/colocation_utils.h"
 #include "distributed/hash_helpers.h"
+#include "distributed/multi_executor.h"
 #include "distributed/multi_join_order.h"
 #include "distributed/multi_partitioning_utils.h"
 #include "distributed/metadata_cache.h"
 #include "distributed/relation_access_tracking.h"
 #include "utils/hsearch.h"
+#include "utils/lsyscache.h"
 
 
+/* Config variables managed via guc.c */
+bool EnforceForeignKeyRestrictions = true;
+
 #define PARALLEL_MODE_FLAG_OFFSET 3
+#define PARALLEL_ACCESS_MASK (int) (0 | \
+									(1 << (PLACEMENT_ACCESS_SELECT + \
+										   PARALLEL_MODE_FLAG_OFFSET)) | \
```

Comment:
we could probably reduce the indentation a bit to improve readability

Reply:
I tried that a bit, but `citus_indent` is constantly produced hard to read things 😄   Added a small comment